client/lib/make-json-schema-parser/test/index.js

```diff
@@ -12,15 +12,17 @@ describe( 'makeJsonSchemaParser', () => {
 
 	test( 'should throw SchemaError on if data does not validate', () => {
 		const parser = makeJsonSchemaParser( { type: 'null' } );
-		expect( () => parser( 0 ) ).toThrow( SchemaError );
+		expect( () => parser( 0 ) ).toThrow( new SchemaError().message );
```

Comment:
I think we could use `toThrowError( SchemaError )` here and avoid having to dig into the schemaError

Reply:
hmm, looks this may be a place where jasmine and jest diverge. https://facebook.github.io/jest/docs/en/expect.html#tothrowerror

Seems like the original signature should have worked?

Reply:
ahhhhhh

so when coverage is enabled, it appears that the typeof check fails, maybe because istanbul is proxying the actual error...

Reply:
```
    Expected the function to throw an error of type:
      "TransformerError"
    Instead, it threw:
      Testing error during transform
          16 | export class TransformerError extends Error {
          17 | 	constructor( error, data, transformer ) {
        > 18 | 		super( error.message );
          19 | 		this.inputData = data;
          20 | 		this.transformer = transformer;
          21 | 	}

          at new TransformerError (client/lib/make-json-schema-parser/index.js:18:3)
          at transform (client/lib/make-json-schema-parser/index.js:120:11)
          at client/lib/make-json-schema-parser/index.js:130:10
          at client/lib/make-json-schema-parser/test/index.js:23:17
          at Object.<anonymous> (node_modules/expect/build/to_throw_matchers.js:42:7)
          at Object.throwingMatcher [as toThrow] (node_modules/expect/build/index.js:147:24)
          at Object.<anonymous> (client/lib/make-json-schema-parser/test/index.js:23:31)

      21 | 		};
      22 | 		const parser = makeJsonSchemaParser( {}, transformer );
    > 23 | 		expect( () => parser( 0 ) ).toThrow( TransformerError );
      24 | 	} );
      25 |
      26 | 	test( 'should return input unchanged if valid and no transformer supplied', () => {

      at Object.<anonymous> (client/lib/make-json-schema-parser/test/index.js:23:31)
```

Reply:
might be this too https://github.com/babel/babel/issues/3083

Reply:
Looks like changing the type out for `Error` and adding the message from the test as the second arg works.

```
        test( 'should throw SchemaError on if data does not validate', () => {
                const parser = makeJsonSchemaParser( { type: 'null' } );
-               expect( () => parser( 0 ) ).toThrow( SchemaError );
+               expect( () => parser( 0 ) ).toThrow( Error, 'Failed to validate with JSON schema' );
        } );
```